## 1. Import Libraries

In [1]:
import warnings
import numpy as np
import pandas as pd
import seaborn as sns
from scipy import stats
import matplotlib.pyplot as plt
from ydata_profiling import ProfileReport

warnings.filterwarnings("ignore")

In [2]:
cab_df = pd.read_csv('/Users/sharonokech/Documents/Data_Glacier_Virtual_Internship/My_Tasks/DataSets/Cab_Data.csv')
cab_df.head(10)
#'/Users/sharonokech/Documents/Data_Glacier_Virtual_Internship/My_Tasks/DataSets/Cab_Data.csv'

Transaction ID Date of Travel   Company        City  KM Travelled  \
0        10000011     2016-01-08  Pink Cab  ATLANTA GA         30.45   
1        10000012     2016-01-06  Pink Cab  ATLANTA GA         28.62   
2        10000013     2016-01-02  Pink Cab  ATLANTA GA          9.04   
3        10000014     2016-01-07  Pink Cab  ATLANTA GA         33.17   
4        10000015     2016-01-03  Pink Cab  ATLANTA GA          8.73   
5        10000016     2016-01-07  Pink Cab  ATLANTA GA          6.06   
6        10000017     2016-01-03  Pink Cab   AUSTIN TX         44.00   
7        10000018     2016-01-07  Pink Cab   AUSTIN TX         35.65   
8        10000019     2016-01-12  Pink Cab   BOSTON MA         14.40   
9        10000020     2016-01-06  Pink Cab   BOSTON MA         10.89   

   Price Charged  Cost of Trip  
0         370.95       313.635  
1         358.52       334.854  
2         125.20        97.632  
3         377.40       351.602  
4         114.62        97.776  
5          72.43        63.024  
6         576.15       475.200  
7         466.10       377.890  
8         191.61       146.880  
9         156.98       113.256

In [3]:
#cabc_df = cab_df.pivot(index='City', columns='Company' , values='Cost of Trip')

## 2. Descriptive Data Analysis
### Consice Summary of the Data Frame


In [4]:
print('DataFrame shape:', cab_df.shape)

print('DataFrame size:', cab_df.size)


DataFrame shape: (359392, 7)
DataFrame size: 2515744


In [5]:
#there no missing values due to the no null
cab_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 359392 entries, 0 to 359391
Data columns (total 7 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   Transaction ID  359392 non-null  int64  
 1   Date of Travel  359392 non-null  object 
 2   Company         359392 non-null  object 
 3   City            359392 non-null  object 
 4   KM Travelled    359392 non-null  float64
 5   Price Charged   359392 non-null  float64
 6   Cost of Trip    359392 non-null  float64
dtypes: float64(3), int64(1), object(3)
memory usage: 19.2+ MB


In [6]:
cab_df.isnull()

Transaction ID  Date of Travel  Company   City  KM Travelled  \
0                False           False    False  False         False   
1                False           False    False  False         False   
2                False           False    False  False         False   
3                False           False    False  False         False   
4                False           False    False  False         False   
...                ...             ...      ...    ...           ...   
359387           False           False    False  False         False   
359388           False           False    False  False         False   
359389           False           False    False  False         False   
359390           False           False    False  False         False   
359391           False           False    False  False         False   

        Price Charged  Cost of Trip  
0               False         False  
1               False         False  
2               False         False  
3               False         False  
4               False         False  
...               ...           ...  
359387          False         False  
359388          False         False  
359389          False         False  
359390          False         False  
359391          False         False  

[359392 rows x 7 columns]

In [7]:
cab_df.sort_values(by=['Price Charged'], ascending=False)


Transaction ID Date of Travel     Company           City  \
138644        10170615     2017-05-05  Yellow Cab    NEW YORK NY   
138669        10170640     2017-05-05  Yellow Cab    NEW YORK NY   
138766        10170737     2017-04-30  Yellow Cab    NEW YORK NY   
247599        10302893     2018-02-16  Yellow Cab  PITTSBURGH PA   
138548        10170519     2017-04-30  Yellow Cab    NEW YORK NY   
...                ...            ...         ...            ...   
285213        10350487     2018-07-23    Pink Cab      DALLAS TX   
132169        10162663     2017-04-17    Pink Cab      DALLAS TX   
15884         10020197     2016-03-30    Pink Cab      BOSTON MA   
246581        10301695     2018-02-20    Pink Cab  PITTSBURGH PA   
143508        10176703     2017-05-29    Pink Cab   NASHVILLE TN   

        KM Travelled  Price Charged  Cost of Trip  
138644         46.80        2048.03      584.0640  
138669         43.29        2016.70      571.4280  
138766         43.20        2013.95      580.6080  
247599         47.20        1993.83      594.7200  
138548         41.81        1981.05      556.9092  
...              ...            ...           ...  
285213          1.94          16.76       19.9820  
132169          1.92          16.53       19.2000  
15884           1.90          16.38       19.3800  
246581          2.10          15.75       24.7800  
143508          1.98          15.60       21.3840  

[359392 rows x 7 columns]

In [8]:
cab_df.dtypes

Transaction ID      int64
Date of Travel     object
Company            object
City               object
KM Travelled      float64
Price Charged     float64
Cost of Trip      float64
dtype: object

In [9]:
cab_df.head

<bound method NDFrame.head of         Transaction ID Date of Travel     Company           City  \
0             10000011     2016-01-08    Pink Cab     ATLANTA GA   
1             10000012     2016-01-06    Pink Cab     ATLANTA GA   
2             10000013     2016-01-02    Pink Cab     ATLANTA GA   
3             10000014     2016-01-07    Pink Cab     ATLANTA GA   
4             10000015     2016-01-03    Pink Cab     ATLANTA GA   
...                ...            ...         ...            ...   
359387        10440101     2018-01-08  Yellow Cab  WASHINGTON DC   
359388        10440104     2018-01-04  Yellow Cab  WASHINGTON DC   
359389        10440105     2018-01-05  Yellow Cab  WASHINGTON DC   
359390        10440106     2018-01-05  Yellow Cab  WASHINGTON DC   
359391        10440107     2018-01-02  Yellow Cab  WASHINGTON DC   

        KM Travelled  Price Charged  Cost of Trip  
0              30.45         370.95      313.6350  
1              28.62         358.52      334.8540

In [10]:
cab_df['Date of Travel'] = pd.to_datetime(cab_df['Date of Travel'])#, format='%Y%m%d')
cab_df.dtypes

Transaction ID             int64
Date of Travel    datetime64[ns]
Company                   object
City                      object
KM Travelled             float64
Price Charged            float64
Cost of Trip             float64
dtype: object

In [11]:
#factor variable in the company column: Pink Cab and Yellow Cab
cab_df['Company'].unique()

array(['Pink Cab', 'Yellow Cab'], dtype=object)

In [12]:
cab_df['City'].unique()

array(['ATLANTA GA', 'AUSTIN TX', 'BOSTON MA', 'CHICAGO IL', 'DALLAS TX',
       'DENVER CO', 'LOS ANGELES CA', 'MIAMI FL', 'NASHVILLE TN',
       'NEW YORK NY', 'ORANGE COUNTY', 'PHOENIX AZ', 'PITTSBURGH PA',
       'SACRAMENTO CA', 'SAN DIEGO CA', 'SEATTLE WA', 'SILICON VALLEY',
       'TUCSON AZ', 'WASHINGTON DC'], dtype=object)

In [13]:
cab_df.describe()

Transaction ID   KM Travelled  Price Charged   Cost of Trip
count    3.593920e+05  359392.000000  359392.000000  359392.000000
mean     1.022076e+07      22.567254     423.443311     286.190113
std      1.268058e+05      12.233526     274.378911     157.993661
min      1.000001e+07       1.900000      15.600000      19.000000
25%      1.011081e+07      12.000000     206.437500     151.200000
50%      1.022104e+07      22.440000     386.360000     282.480000
75%      1.033094e+07      32.960000     583.660000     413.683200
max      1.044011e+07      48.000000    2048.030000     691.200000

profile = ProfileReport(cab_df,title="Cab Data Profile"
profile.to_widgets()

In [14]:

Transaction_ID_df = pd.read_csv('/Users/sharonokech/Documents/Data_Glacier_Virtual_Internship/My_Tasks/DataSets/Transaction_ID.csv')
Transaction_ID_df.head(10)

Transaction ID  Customer ID Payment_Mode
0        10000011        29290         Card
1        10000012        27703         Card
2        10000013        28712         Cash
3        10000014        28020         Cash
4        10000015        27182         Card
5        10000016        27318         Cash
6        10000017        33788         Card
7        10000018        34106         Card
8        10000019        59799         Cash
9        10000020        57982         Cash

In [15]:
print('DataFrame shape:', Transaction_ID_df.shape)

print('DataFrame size:', Transaction_ID_df.size)

DataFrame shape: (440098, 3)
DataFrame size: 1320294


In [16]:
Transaction_ID_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 440098 entries, 0 to 440097
Data columns (total 3 columns):
 #   Column          Non-Null Count   Dtype 
---  ------          --------------   ----- 
 0   Transaction ID  440098 non-null  int64 
 1   Customer ID     440098 non-null  int64 
 2   Payment_Mode    440098 non-null  object
dtypes: int64(2), object(1)
memory usage: 10.1+ MB


In [17]:
Transaction_ID_df.dtypes

Transaction ID     int64
Customer ID        int64
Payment_Mode      object
dtype: object

In [18]:
Transaction_ID_df.isnull()

Transaction ID  Customer ID  Payment_Mode
0                False        False         False
1                False        False         False
2                False        False         False
3                False        False         False
4                False        False         False
...                ...          ...           ...
440093           False        False         False
440094           False        False         False
440095           False        False         False
440096           False        False         False
440097           False        False         False

[440098 rows x 3 columns]

In [19]:
Transaction_ID_df['Payment_Mode'].unique()

array(['Card', 'Cash'], dtype=object)

In [20]:
#Change Payment_Mode to category type
Transaction_ID_df['Payment_Mode'] = Transaction_ID_df['Payment_Mode'].astype('category')
Transaction_ID_df.dtypes

Transaction ID       int64
Customer ID          int64
Payment_Mode      category
dtype: object

In [21]:
#checking for duplicates in colums
Transaction_ID_df['Transaction ID'].duplicated()

0         False
1         False
2         False
3         False
4         False
          ...  
440093    False
440094    False
440095    False
440096    False
440097    False
Name: Transaction ID, Length: 440098, dtype: bool

In [22]:
#Customer column has duplicates implying one customer can own several transaction IDS.
Transaction_ID_df['Customer ID'].duplicated()

0         False
1         False
2         False
3         False
4         False
          ...  
440093     True
440094     True
440095     True
440096     True
440097     True
Name: Customer ID, Length: 440098, dtype: bool

In [23]:
#Transaction_ID_df.describe

In [24]:
city_df= pd.read_csv('/Users/sharonokech/Documents/Data_Glacier_Virtual_Internship/My_Tasks/DataSets/city.csv')
city_df.head(10)

City   Population      Users
0     NEW YORK NY   8,405,837    302,149 
1      CHICAGO IL   1,955,130    164,468 
2  LOS ANGELES CA   1,595,037    144,132 
3        MIAMI FL   1,339,155     17,675 
4  SILICON VALLEY   1,177,609     27,247 
5   ORANGE COUNTY   1,030,185     12,994 
6    SAN DIEGO CA     959,307     69,995 
7      PHOENIX AZ     943,999      6,133 
8       DALLAS TX     942,908     22,157 
9      ATLANTA GA     814,885     24,701

In [25]:
city_df['Population'] = city_df['Population'].str.replace(',', '')
city_df['Population']= city_df['Population'].astype('int')

In [26]:
city_df['Users']= city_df['Users'].str.replace(',', '')
city_df['Users']= city_df['Users'].astype('int')

In [27]:
city_df.dtypes


City          object
Population     int64
Users          int64
dtype: object

In [28]:
print('DataFrame shape:', city_df.shape)

print('DataFrame size:', city_df.size)

DataFrame shape: (20, 3)
DataFrame size: 60


In [29]:
city_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20 entries, 0 to 19
Data columns (total 3 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   City        20 non-null     object
 1   Population  20 non-null     int64 
 2   Users       20 non-null     int64 
dtypes: int64(2), object(1)
memory usage: 608.0+ bytes


city_df.isnull()

city_df['City'].unique

In [30]:
city_df.describe()
# Mean users are 64520

Population          Users
count  2.000000e+01      20.000000
mean   1.231592e+06   64520.650000
std    1.740127e+06   83499.375289
min    2.489680e+05    3643.000000
25%    6.086372e+05   11633.250000
50%    7.845590e+05   23429.000000
75%    1.067041e+06   91766.000000
max    8.405837e+06  302149.000000

In [31]:
#CustomerID Data
Customer_ID_df = pd.read_csv('/Users/sharonokech/Documents/Data_Glacier_Virtual_Internship/My_Tasks/DataSets/Customer_ID.csv')
Customer_ID_df.head(10)

Customer ID Gender  Age  Income (USD/Month)
0        29290   Male   28               10813
1        27703   Male   27                9237
2        28712   Male   53               11242
3        28020   Male   23               23327
4        27182   Male   33                8536
5        27318   Male   25               13984
6        33788   Male   23               23788
7        34106   Male   19               19980
8        59799   Male   33               19271
9        57982   Male   57                5068

In [32]:
print('DataFrame shape:', Customer_ID_df.shape)

print('DataFrame size:', Customer_ID_df.size)

DataFrame shape: (49171, 4)
DataFrame size: 196684


In [33]:
Customer_ID_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 49171 entries, 0 to 49170
Data columns (total 4 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   Customer ID         49171 non-null  int64 
 1   Gender              49171 non-null  object
 2   Age                 49171 non-null  int64 
 3   Income (USD/Month)  49171 non-null  int64 
dtypes: int64(3), object(1)
memory usage: 1.5+ MB


In [34]:
Customer_ID_df['Gender'] = Customer_ID_df['Gender'].astype('category')
Customer_ID_df.dtypes


Customer ID              int64
Gender                category
Age                      int64
Income (USD/Month)       int64
dtype: object

In [35]:


Customer_ID_df['Customer ID'].unique

<bound method Series.unique of 0        29290
1        27703
2        28712
3        28020
4        27182
         ...  
49166    12490
49167    14971
49168    41414
49169    41677
49170    39761
Name: Customer ID, Length: 49171, dtype: int64>

In [36]:
Customer_ID_df.describe(include='all')

Customer ID Gender           Age  Income (USD/Month)
count   49171.000000  49171  49171.000000        49171.000000
unique           NaN      2           NaN                 NaN
top              NaN   Male           NaN                 NaN
freq             NaN  26562           NaN                 NaN
mean    28398.252283    NaN     35.363121        15015.631856
std     17714.137333    NaN     12.599066         8002.208253
min         1.000000    NaN     18.000000         2000.000000
25%     12654.500000    NaN     25.000000         8289.500000
50%     27631.000000    NaN     33.000000        14656.000000
75%     43284.500000    NaN     42.000000        21035.000000
max     60000.000000    NaN     65.000000        35000.000000

# Correlation Analysis

In [37]:
Merged_df= cab_df.merge(Transaction_ID_df, on= 'Transaction ID').merge(city_df, on = 'City').merge(Customer_ID_df, on ='Customer ID')
Merged_df.head(10)

Transaction ID Date of Travel     Company        City  KM Travelled  \
0        10000011     2016-01-08    Pink Cab  ATLANTA GA         30.45   
1        10351127     2018-07-21  Yellow Cab  ATLANTA GA         26.19   
2        10412921     2018-11-23  Yellow Cab  ATLANTA GA         42.55   
3        10000012     2016-01-06    Pink Cab  ATLANTA GA         28.62   
4        10320494     2018-04-21  Yellow Cab  ATLANTA GA         36.38   
5        10324737     2018-05-04  Yellow Cab  ATLANTA GA          6.18   
6        10395626     2018-10-27    Pink Cab  ATLANTA GA         13.39   
7        10000013     2016-01-02    Pink Cab  ATLANTA GA          9.04   
8        10079404     2016-09-21  Yellow Cab  ATLANTA GA         39.60   
9        10186994     2017-06-23  Yellow Cab  ATLANTA GA         18.19   

   Price Charged  Cost of Trip  Customer ID Payment_Mode  Population  Users  \
0         370.95      313.6350        29290         Card      814885  24701   
1         598.70      317.4228        29290         Cash      814885  24701   
2         792.05      597.4020        29290         Card      814885  24701   
3         358.52      334.8540        27703         Card      814885  24701   
4         721.10      467.1192        27703         Card      814885  24701   
5         138.40       87.5088        27703         Cash      814885  24701   
6         167.03      141.9340        27703         Card      814885  24701   
7         125.20       97.6320        28712         Cash      814885  24701   
8         704.30      494.2080        28712         Card      814885  24701   
9         365.63      246.6564        28712         Card      814885  24701   

  Gender  Age  Income (USD/Month)  
0   Male   28               10813  
1   Male   28               10813  
2   Male   28               10813  
3   Male   27                9237  
4   Male   27                9237  
5   Male   27                9237  
6   Male   27                9237  
7   Male   53               11242  
8   Male   53               11242  
9   Male   53               11242

In [38]:
Merged_df.isnull()

Transaction ID  Date of Travel  Company   City  KM Travelled  \
0                False           False    False  False         False   
1                False           False    False  False         False   
2                False           False    False  False         False   
3                False           False    False  False         False   
4                False           False    False  False         False   
...                ...             ...      ...    ...           ...   
359387           False           False    False  False         False   
359388           False           False    False  False         False   
359389           False           False    False  False         False   
359390           False           False    False  False         False   
359391           False           False    False  False         False   

        Price Charged  Cost of Trip  Customer ID  Payment_Mode  Population  \
0               False         False        False         False       False   
1               False         False        False         False       False   
2               False         False        False         False       False   
3               False         False        False         False       False   
4               False         False        False         False       False   
...               ...           ...          ...           ...         ...   
359387          False         False        False         False       False   
359388          False         False        False         False       False   
359389          False         False        False         False       False   
359390          False         False        False         False       False   
359391          False         False        False         False       False   

        Users  Gender    Age  Income (USD/Month)  
0       False   False  False               False  
1       False   False  False               False  
2       False   False  False               False  
3       False   False  False               False  
4       False   False  False               False  
...       ...     ...    ...                 ...  
359387  False   False  False               False  
359388  False   False  False               False  
359389  False   False  False               False  
359390  False   False  False               False  
359391  False   False  False               False  

[359392 rows x 14 columns]

In [39]:
Merged_df.dtypes

Transaction ID                 int64
Date of Travel        datetime64[ns]
Company                       object
City                          object
KM Travelled                 float64
Price Charged                float64
Cost of Trip                 float64
Customer ID                    int64
Payment_Mode                category
Population                     int64
Users                          int64
Gender                      category
Age                            int64
Income (USD/Month)             int64
dtype: object

profile = ProfileReport(Merged_df,title="Carb Merged Data Profile")
profile.to_widgets()

profile = ProfileReport(Merged_df,title="Carb Merged Data Profile")
profile.to_widgets()

## Correlation

In [40]:
Merged_df.describe()

Transaction ID   KM Travelled  Price Charged   Cost of Trip  \
count    3.593920e+05  359392.000000  359392.000000  359392.000000   
mean     1.022076e+07      22.567254     423.443311     286.190113   
std      1.268058e+05      12.233526     274.378911     157.993661   
min      1.000001e+07       1.900000      15.600000      19.000000   
25%      1.011081e+07      12.000000     206.437500     151.200000   
50%      1.022104e+07      22.440000     386.360000     282.480000   
75%      1.033094e+07      32.960000     583.660000     413.683200   
max      1.044011e+07      48.000000    2048.030000     691.200000   

         Customer ID    Population          Users            Age  \
count  359392.000000  3.593920e+05  359392.000000  359392.000000   
mean    19191.652115  3.132198e+06  158365.582267      35.336705   
std     21012.412463  3.315194e+06  100850.051020      12.594234   
min         1.000000  2.489680e+05    3643.000000      18.000000   
25%      2705.000000  6.712380e+05   80021.000000      25.000000   
50%      7459.000000  1.595037e+06  144132.000000      33.000000   
75%     36078.000000  8.405837e+06  302149.000000      42.000000   
max     60000.000000  8.405837e+06  302149.000000      65.000000   

       Income (USD/Month)  
count       359392.000000  
mean         15048.822937  
std           7969.409482  
min           2000.000000  
25%           8424.000000  
50%          14685.000000  
75%          21035.000000  
max          35000.000000

In [41]:
Merged_df.corr(method ='pearson')

Transaction ID  KM Travelled  Price Charged  Cost of Trip  \
Transaction ID            1.000000     -0.001429      -0.052902     -0.003462   
KM Travelled             -0.001429      1.000000       0.835753      0.981848   
Price Charged            -0.052902      0.835753       1.000000      0.859812   
Cost of Trip             -0.003462      0.981848       0.859812      1.000000   
Customer ID              -0.016912      0.000389      -0.177324      0.003077   
Population                0.023868     -0.002311       0.326589      0.015108   
Users                     0.013526     -0.000428       0.281061      0.023628   
Age                      -0.001267     -0.000369      -0.003084     -0.000189   
Income (USD/Month)       -0.001570     -0.000544       0.003228     -0.000633   

                    Customer ID  Population     Users       Age  \
Transaction ID        -0.016912    0.023868  0.013526 -0.001267   
KM Travelled           0.000389   -0.002311 -0.000428 -0.000369   
Price Charged         -0.177324    0.326589  0.281061 -0.003084   
Cost of Trip           0.003077    0.015108  0.023628 -0.000189   
Customer ID            1.000000   -0.647052 -0.610742 -0.004735   
Population            -0.647052    1.000000  0.915490 -0.009002   
Users                 -0.610742    0.915490  1.000000 -0.005906   
Age                   -0.004735   -0.009002 -0.005906  1.000000   
Income (USD/Month)    -0.013608    0.011868  0.010464  0.003907   

                    Income (USD/Month)  
Transaction ID               -0.001570  
KM Travelled                 -0.000544  
Price Charged                 0.003228  
Cost of Trip                 -0.000633  
Customer ID                  -0.013608  
Population                    0.011868  
Users                         0.010464  
Age                           0.003907  
Income (USD/Month)            1.000000

In [42]:
Merged_df.corr().style.background_gradient(cmap='RdYlGn', axis=None)

### Observations:
Highest Correlation exist between: 


                                   KM Travelled and Cost of Trip= '0.981848', 
                                   Cost of Trip and Price Charged= '0.835753', and  
                                   KM Travelled and Price Charged= '0.835753'
Followed by:                       
                                   
                                   Customer_ID and the Price Charged

### Graphical Correlation Analysis

#Merged_df[['KM Travelled', 'Price Charged', 'Cost of Trip','Age','Income (USD/Month)']].plot.hist(alpha=0.5);
from pandas.plotting import scatter_matrix
scatter_matrix(Merged_df[['KM Travelled','Price Charged','Cost of Trip','Age','Income (USD/Month)' ]],figsize=(10, 10),);

sns.pairplot(Merged_df, hue='Company', height=1.5)

## 3. Data Examination
###  Outlier Identification

 
fig = plt.figure(figsize =(7, 10))
#plt.rcParams["figure.autolayout"] = True
Merged_boxplot= Merged_df.boxplot(column=['KM Travelled', 'Price Charged', 'Cost of Trip','Age','Income (USD/Month)'])
#price_charged has outliers

Merged_df.hist()
fig = plt.figure(figsize =(10, 10))

### Price Charged


Price_charged_scatter = Merged_df.plot.scatter(y='Price Charged',
...                   x='Date of Travel',
...                   c='DarkBlue')

Cost_of_Trip_scatter = Merged_df.plot.scatter(y='Cost of Trip',
...                   x='Date of Travel',
...                   c='Red')

    The numerical histograms for 'Date of Travel', KM Travelled, Price Charged, Cost of Trip, Age and Income(USD/Month), are right skewed thus mean> median > mode.
    Outliers cannot be eliminated using z value but Interquartile range.
   

##### Removing Outliers 

In [43]:

upper_limit = Merged_df["Price Charged"].mean() + 3*Merged_df["Price Charged"].std()

print("Upper limit:", upper_limit)

lower_limit = Merged_df["Price Charged"].mean()-3*Merged_df["Price Charged"].std()

print("lower limit:", lower_limit)

Merged_df["Price Charged"] = np.where(Merged_df["Price Charged"] > upper_limit,

    upper_limit,

    np.where (

       Merged_df["Price Charged"] < lower_limit,

       lower_limit,

       Merged_df["Price Charged"]

   )

)




Upper limit: 1246.5800455975784
lower limit: -399.6934230795479


Outliers[]
MergedOut_df=Merged_df
Outliers[]=Merged_df["Price Charged"]-MergedOut_df["Price Charged"]
Outliers


In [44]:
Merged_df.describe()[["Price Charged"]]

Price Charged
count  359392.000000
mean      422.281094
std       270.344818
min        15.600000
25%       206.437500
50%       386.360000
75%       583.660000
max      1246.580046

Price_charged_scatter = Merged_df.plot.scatter(x='Date of Travel',
...                   y= 'Price Charged',
...                   c='DarkBlue')

In [ ]:
l_df = pd.DataFrame(Merged_df, columns=["Users", "Price Charged", "KM Travelled","Date of Travel"])
 
l_df.plot(x="Date of Travel", y=["Users", "Price Charged", "KM Travelled"], kind="bar", figsize=(9, 8))
plt.show()


### KM Travelled


####  Payment Mode

In [ ]:
plt.figure(figsize = (16, 9))
ax = sns.countplot(x="Company", hue="Payment_Mode", data=Transaction_ID_df)
plt.title('Payment Mode in both companies', fontsize=19)
plt.show()

#### Gender

## City

In [ ]:
no_of_users = (city_df['Users'] /city_df['Population']) * 100 
cities = city_df['City']
from random import randint

colors = []
n = 16

for i in range(n):
    colors.append('#%06X' % randint(0, 0xFFFFFF))
plt.figure(figsize = (16, 9))
plt.bar(cities, no_of_users, edgecolor='black', color = colors)
plt.gcf().autofmt_xdate()
plt.title('User Percentage of Population by Cities')
plt.ylabel('Percentage (%)')
plt.xlabel('Cities')
plt.show()

In [ ]:
plot = city_df.plot.pie(subplots=True, figsize=(5, 5))

In [ ]:
sns.countplot(data=df, x="class", hue="alive")

In [ ]:
Merged_df.plot(y='Population', x='Users', kind='pie', labels=Merged_df['City'], title='Population by City', subplots=True)
plt.show()


# Hypothesis and  Investigate

###### Hypothesis 1: Do users prefer cash or Card Payments?

In [ ]:
H0: Cab users generally prefer using Card Payments to Cash Payments.
H1: Cab users generally do notprefer using Card Payments to Cash Payments.